### PR TITLE
Proposal time and enactment time edgecases (2+1 blocks at least)

### PR DESCRIPTION
**Pull request by junkai121**

### CODE REVIEW

File: proto/kira/gov/network_properties.proto [hunk 1]
```diff
@@ -20,35 +20,39 @@ enum NetworkProperty {
...
     PROPOSAL_ENACTMENT_TIME = 4 [(gogoproto.enumvalue_customname) = "ProposalEnactmentTime"];
-    ENABLE_FOREIGN_TX_FEE_PAYMENTS = 5 [(gogoproto.enumvalue_customname) = "EnableForeignFeePayments"];
-    MISCHANCE_RANK_DECREASE_AMOUNT = 6 [(gogoproto.enumvalue_customname) = "MischanceRankDecreaseAmount"];
-    INACTIVE_RANK_DECREASE_PERCENT = 7 [(gogoproto.enumvalue_customname) = "InactiveRankDecreasePercent"];
-    POOR_NETWORK_MAX_BANK_SEND = 8 [(gogoproto.enumvalue_customname) = "PoorNetworkMaxBankSend"];
-    MIN_VALIDATORS = 9 [(gogoproto.enumvalue_customname) = "MinValidators"];
-    JAIL_MAX_TIME = 10 [(gogoproto.enumvalue_customname) = "JailMaxTime"];
-    ENABLE_TOKEN_WHITELIST = 11 [(gogoproto.enumvalue_customname) = "EnableTokenWhitelist"];
-    ENABLE_TOKEN_BLACKLIST = 12 [(gogoproto.enumvalue_customname) = "EnableTokenBlacklist"];
+    MIN_PROPOSAL_END_BLOCKS = 5 [(gogoproto.enumvalue_customname) = "MinProposalEndBlocks"];
```
Comment (rgrcoding) [inline proto/kira/gov/network_properties.proto:23]: We need to start to keep the numbers as they were, in protocol buffers in order to be compatible with changes, the number after the equal should not be changed. Now maybe is not a problem, but this in production means the chain is broken.

File: proto/kira/gov/network_properties.proto [hunk 2]
```diff
@@ -20,35 +20,39 @@ enum NetworkProperty {
...
     // Every time node status changes to inactive the rank should be divided by 2, that is rank = FLOOR(rank / 2)
     // The streak and rank will enable governance to judge real life performance of validators on the mainnet or testnet, and potentially propose eviction of the weakest and least reliable operators.
-    uint64 mischance_rank_decrease_amount = 7; // X (default 10)
-    uint64 inactive_rank_decrease_percent = 8; // Y (default 50%)
-    uint64 min_validators = 9;
-    uint64 poor_network_max_bank_send = 10; // default 10000 ukex
-    uint64 jail_max_time = 11; // Jailing validator maximum time in minutes
-    bool enable_token_whitelist = 12; // TokenWhitelist is valid when this param is set
-    bool enable_token_blacklist = 13; // TokenBlacklist is valid when this param is set
+    uint64 mischance_rank_decrease_amount = 9; // X (default 10)
```
Comment (rgrcoding) [inline proto/kira/gov/network_properties.proto:51]: Same. https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers

File: proto/kira/gov/proposal.proto [hunk 3]
```diff
@@ -109,14 +109,18 @@ message Proposal {
   [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"submit_time\""];
 
   google.protobuf.Timestamp voting_end_time = 4
-  [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"voting_end_time\""];
+    [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"voting_end_time\""];
 
   google.protobuf.Timestamp enactment_end_time = 5
   [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"enactment_end_time\""];
 
-  VoteResult result = 6;
```
Comment (rgrcoding) [inline proto/kira/gov/proposal.proto:117]: Same.

File: proto/kira/gov/network_properties.proto (hunk 1 above)
Comment (junkai121) [reply proto/kira/gov/network_properties.proto:23]: yes, for now, we need to make things organized with numbers before going production.

**Pull request closed without merging**